examples/step-5/step-5.cc

```diff
@@ -113,6 +113,14 @@ double coefficient (const Point<dim> &p)
   else
     return 1;
 }
+// @note It looks like we use a discontinuous coefficient, however,
+// the finite element method using quadrature really sees only a
+// polynomial approximation (defined by a
+// <code>quadrature_formula</code>) of that discontinuous coefficient
+// on every mesh. (Though that polynomial approximation of course
+// tends to a discontinuous function under mesh refinement.) Refer to
+// CellAccessor::set_material_id to set a discontinuous coefficient
```

Comment:
Consider a quadrature point located on the boundary of two cells with material1 and material2. In the weak form, you will evaluate two different values of the coefficient (which can be equal to each other, but in general are not) at this point depending on the material ID when assembling the local matrix on each of this two cells.  It is for sure will correspond a discontinuity in global system matrix. At the same time, you are right, it is not possible to obtain a discontinuity inside of the element due to the quadrature approximation of the integral. See original thread in google groups for more details https://groups.google.com/forum/#!topic/dealii/J8rQLvbO7YQ